zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -162,6 +170,46 @@ def test_fail_request(self) -> None:
         assert bot_user.bot_owner is not None
         self.assertEqual(bot_owner_notification.recipient_id, bot_user.bot_owner.recipient_id)
 
+    def test_fail_request_with_deactivated_bot(self) -> None:
+        bot_user = self.example_user("outgoing_webhook_bot")
+        mock_event = self.mock_event(bot_user)
+        service_handler = GenericOutgoingWebhookService("token", bot_user, "service")
+
+        def deactivate_bot_first(*args: Any) -> None:
+            bot_user = get_user_profile_by_id(args[0]["user_profile_id"])
+            do_deactivate_user(bot_user, acting_user=None)
+            fail_with_message(*args)
+
+        expected_fail = mock.patch(
+            "zerver.lib.outgoing_webhook.fail_with_message", side_effect=deactivate_bot_first
+        )
+
+        with mock.patch.object(
+            service_handler, "session"
+        ) as session, expected_fail as mock_fail, self.assertLogs(level="WARNING") as m:
+            session.post.return_value = ResponseMock(408)
+            final_response = do_rest_call("", mock_event, service_handler)
+            assert final_response is not None
+
+            self.assertEqual(
+                m.output,
+                [
+                    f'WARNING:root:Message http://zulip.testserver/#narrow/stream/999-Verona/topic/Foo/near/ triggered an outgoing webhook, returning status code 408.\n Content of response (in quotes): "{final_response.text}"'
+                ],
+            )
+
+        self.assertTrue(mock_fail.called)
+
+        bot_owner_notification = self.get_last_message()
+        self.assertEqual(
+            bot_owner_notification.content,
+            """[A message](http://zulip.testserver/#narrow/stream/999-Verona/topic/Foo/near/) to your bot @_**Outgoing Webhook** triggered an outgoing webhook.
+The webhook got a response with status code *408*.""",
+        )
+
+        assert bot_user.bot_owner is not None
+        self.assertEqual(bot_owner_notification.recipient_id, bot_user.bot_owner.recipient_id)
```

Comment:
These asserts seem to be in the wrong order; we should probably assert that the bot has an owner at the very top.

Reply:
Makes sense. Will do.

Reply:
@timabbott Updated the assert statement for the first commit. For the second commit, the current position of the assert statement is necessary otherwise it throws a mypy error:
```
mypy      | zerver/tests/test_outgoing_webhook_system.py:222: error: Argument 1 to "do_deactivate_user" has incompatible type "Optional[UserProfile]"; expected "UserProfile"  [arg-type]
mypy      | 
mypy      | See https://zulip.readthedocs.io/en/latest/testing/mypy.html for debugging tips.
```

Reply:
ok.